node/grpc/metrics_v2.go

```diff
@@ -0,0 +1,175 @@
+package grpc
+
+import (
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	grpcprom "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"google.golang.org/grpc"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+	"time"
+)
+
+const namespace = "eigenda_node"
+
+// MetricsV2 encapsulates metrics for the v2 DA node.
+type MetricsV2 struct {
+	logger logging.Logger
+
+	registry         *prometheus.Registry
+	grpcServerOption grpc.ServerOption
+
+	storeChunksLatency  *prometheus.SummaryVec
+	storeChunksDataSize *prometheus.GaugeVec
+
+	getChunksLatency  *prometheus.SummaryVec
+	getChunksDataSize *prometheus.GaugeVec
+
+	dbSize           *prometheus.GaugeVec
+	dbSizePollPeriod time.Duration
+	dbDir            string
+	isAlive          *atomic.Bool
+}
+
+// NewV2Metrics creates a new MetricsV2 instance. dbSizePollPeriod is the period at which the database size is polled.
+// If set to 0, the database size is not polled.
+func NewV2Metrics(
+	logger logging.Logger,
+	registry *prometheus.Registry,
+	dbDir string,
+	dbSizePollPeriod time.Duration) (*MetricsV2, error) {
+
+	// These should be re-enabled once the legacy v1 metrics are removed.
+	//registry.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	//registry.MustRegister(collectors.NewGoCollector())
+
+	grpcMetrics := grpcprom.NewServerMetrics()
+	registry.MustRegister(grpcMetrics)
+	grpcServerOption := grpc.UnaryInterceptor(
+		grpcMetrics.UnaryServerInterceptor(),
+	)
+
+	storeChunksLatency := promauto.With(registry).NewSummaryVec(
```

Comment:
We may not need these primitive level metrics as gRPC already provides (need to integrate / enable it)